File: 백준/Silver/8979. 올림픽/올림픽.py
N, need = map(int, input().split())

lst = [list(map(int, input().split())) for _ in range(N)]

# lst[0] : 국가번호
# lst[1] : 금
# lst[2] : 은
# lst[3] : 동
# print(lst)
lst.sort(key = lambda x: (x[1], x[2], x[3]), reverse=True)
# print(lst)
for k in range(N):
    if need == lst[k][0]:
        ans = k

# 만약 금메달 수가 같으면, 은메달 수를 확인
# 은메달 수도 같으면, 동메달 수를 확인
# 동메달도 같다면 앞 등수와 같은 등수 추가
# 그 다음 등수는 len(arr)의 등수로 추가

arr = []
cnt = 1
for i in range(N-1):
    if lst[i][1] == lst[i+1][1] and \
        lst[i][2] == lst[i+1][2] and \
        lst[i][3] == lst[i+1][3]:
        arr.append(cnt)

    else:
        arr.append(cnt)
        cnt += arr.count(cnt)
arr.append(cnt)

for j in range(N):
    if j == ans:
        print(arr[j])

File: 백준/Silver/8979. 올림픽/올림픽.js
const fs = require("fs");
const input = fs.readFileSync("/dev/stdin").toString().trim().split("\n");

const [N, K] = input[0].split(" ").map(Number);
let arr = input.slice(1, N + 1);

// 1부터 N까지의 국가들의 index[1] index[2] index[3] 을 차례로 비교해서 sort
arr = arr.sort((a, b) => {
  const aValues = a.split(" ").map(Number);
  const bValues = b.split(" ").map(Number);

  if (aValues[1] !== bValues[1]) {
    return bValues[1] - aValues[1];
  } else if (aValues[2] !== bValues[2]) {
    return bValues[2] - aValues[2];
  } else {
    return bValues[3] - aValues[3];
  }
});

function count(lst, target) {
  return lst.filter((item) => item === target).length;
}

let lst = [];
cnt = 1;
for (let i = 0; i < N - 1; i++) {
  if (
    arr[i][1] === arr[i + 1][1] &&
    arr[i][2] === arr[i + 1][2] &&
    arr[i][3] === arr[i + 1][3]
  ) {
    lst.push(cnt);
  } else {
    lst.push(cnt);
    cnt += count(lst, cnt);
  }
}
lst.push(cnt);

for (let j = 0; j < N; j++) {
  if (j === K) {
    console.log(lst[j]);
  }
}
